extended/src/main/java/apoc/ml/bedrock/Bedrock.java

```diff
@@ -0,0 +1,123 @@
+package apoc.ml.bedrock;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import apoc.Description;
+import apoc.result.ObjectResult;
+import apoc.util.ExtendedUtil;
+import apoc.util.Util;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.apache.http.impl.client.CloseableHttpClient;
+
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.neo4j.procedure.Name;
+import org.neo4j.procedure.Procedure;
+
+import static apoc.ml.bedrock.BedrockInvokeConfig.MODEL_ID;
+import static apoc.util.JsonUtil.OBJECT_MAPPER;
+import static apoc.ml.bedrock.BedrockInvokeResult.*;
+import static apoc.ml.bedrock.BedrockUtil.*;
+
+
+public class Bedrock {
+    
+    @Procedure
+    public Stream<ModelItemResult> list(@Name(value = "config", defaultValue = "{}") Map<String, Object> config) {
+
+        BedrockConfig conf = new BedrockGetModelsConfig(config);
+        
+        return executeRequestCommon(null, "modelSummaries[*]", conf)
+                .flatMap(i -> ((List<Map<String, Object>>) i).stream())
+                .map(ModelItemResult::new);
+    }
+    
+    @Procedure
+    @Description("To create a customizable bedrock call")
+    public Stream<ObjectResult> custom(@Name(value = "body") Object body,
+                                       @Name(value = "config", defaultValue = "{}") Map<String, Object> config) {
+        
+        return executeCustomRequest(body, config, null)
+                .map(ObjectResult::new);
+    }
+    
+    @Procedure
+    public Stream<Jurassic> jurassic(@Name(value = "body") Object body,
```

Comment:
We should not tie the result type to the model if we can avoid it but convert it to `MapResult`

Reply:
same below for the other procedures, which should be generalized anyhow